pkg/storegateway/series_refs.go

```diff
@@ -1012,7 +1012,7 @@ func (s *loadingSeriesChunkRefsSetIterator) symbolizedSet(ctx context.Context, p
 			}
 		case !s.strategy.isNoChunkRefs():
 			clampLastChunkLength(symbolizedSet.series, metas)
-			series.chunksRanges = metasToRanges([][]chunks.Meta{metas}, s.blockID, s.minTime, s.maxTime)
+			series.chunksRanges = metasToRanges(partitionChunks(metas, 1, 1), s.blockID, s.minTime, s.maxTime)
```

Comment:
I was planning to remove partitionChunks and chunks ranges altogether in a future PR. I wanted to do that this week, but this bug took too much time to find and fix, so it will probably roll into next week. Should I still simplify partitionChunks before that its removal?

Reply:
>Should I still simplify partitionChunks before that its removal?

No :)